packages/babel-generator/src/generators/expressions.ts

```diff
@@ -81,21 +81,29 @@ export function NewExpression(
   if (
     this.format.minified &&
     node.arguments.length === 0 &&
-    !node.optional &&
     !isCallExpression(parent, { callee: node }) &&
     !isMemberExpression(parent) &&
     !isNewExpression(parent)
   ) {
-    return;
+    if (!process.env.BABEL_8_BREAKING) {
+      // @ts-ignore(Babel 7 vs Babel 8) Babel 7 AST
+      if (!node.optional) return;
+    } else {
+      return;
+    }
   }
 
   this.print(node.typeArguments, node); // Flow
   this.print(node.typeParameters, node); // TS
 
-  if (node.optional) {
-    // TODO: This can never happen
-    this.token("?.");
+  if (!process.env.BABEL_8_BREAKING) {
+    // @ts-ignore(Babel 7 vs Babel 8) Babel 7 AST
+    if (node.optional) {
+      // TODO: This can never happen
+      this.token("?.");
```

Comment:
If this can never happen, can we just delete this case? `new a?.()` is invalid syntax.

Reply:
Sorry I wasn't clear on this...
But since you said it, I'll remove it.